lib/cocoapods/installer/pod_source_installer.rb

```diff
@@ -117,13 +117,29 @@ def unlock_files!(file_accessors)
       # @return [void]
       #
       def download_source
+        verify_source_is_secure(root_spec)
         download_result = Downloader.download(download_request, root, :can_cache => can_cache?)
 
         if (@specific_source = download_result.checkout_options) && specific_source != root_spec.source
           sandbox.store_checkout_source(root_spec.name, specific_source)
         end
       end
 
+      # Verify the source of the spec is secure, which is used to
+      # show a warning to the user if that isn't the case
+      # This method doesn't verify all protocols, but currently
+      # only prohibits unencrypted http:// connections
+      #
+      def verify_source_is_secure(root_spec)
+        return if root_spec.source.nil? || root_spec.source[:http].nil?
+        http_source = root_spec.source[:http]
+        return if http_source.start_with?("https://")
+
+        UI.warn "'#{root_spec.name}' uses the unencrypted http protocol to transfer the Pod. " \
```

Comment:
my guess is this unconditionally warning might bother a bunch of people?

Reply:
Yes, I think that's kind of the idea, similar to what chrome does [with the `Not Secure` badge](https://developers.google.com/web/updates/2016/10/avoid-not-secure-warn), this way the users/customers of an SDK will also add pressure to the SDK providers to fix their infrastructure and get a (free, or cheap) SSL certificate.

Reply:
I personally don't like that.

Reply:
could we possibly go through an installation option that is on by default?

Reply:
I'm personally fine with this for now, and getting more strict in the future per #7260. With the change proposed in #7260, we can get rid of this warning, as consumers have explicitly opted in.